core/src/main/java/org/kiwix/kiwixmobile/core/search/DefaultAdapter.java

```diff
@@ -0,0 +1,85 @@
+/*
+ * Kiwix Android
+ * Copyright (c) 2019 Kiwix <android.kiwix.org>
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package org.kiwix.kiwixmobile.core.search;
+
+import android.content.Context;
+import android.text.Html;
+import android.text.style.ClickableSpan;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.TextView;
+import androidx.annotation.NonNull;
+import androidx.recyclerview.widget.RecyclerView;
+import java.util.List;
+import javax.annotation.Nonnull;
+
+public class DefaultAdapter extends RecyclerView.Adapter<DefaultAdapter.ViewHolder> {
+
+  private Context context;
+  List<String> searchList;
+
+  interface CLickListener {
+    void onItemClick(String text);
+    boolean onItemLongClick(String text);
```

Comment:
this doesn't need to return a boolean.
Boolean is the return type of View.OnLongClickListener because the view wants to know if the click event was handled or not, we don't care about that and can just always return true where we set the longClickListener, this means we can focus on the action we want to perform in our implementation

Reply:
Initially, I was not using boolean but that kept on showing me the error:
```
bad return type in lambda expression, void cannot be converted to boolean
```

Reply:
that is because you are using an `expression lambda`, you will need to use a `statement lambda`
eg
```
v.setOnLongClickListener(v->{
listener.onLongClick()
return true
)
```
if you ever have trouble with lambdas you can try to write out the longform, remember that lambdas are only syntax sugar
eg
```
v.setOnLongClickListener(new View.OnLongClickListener(){
@Override public Boolean onLongClick(some arguments or something){
   listener.onLongClick()
  return true
}
})
```
and then studio will give you a prompt to convert it to a lambda